luaui/Widgets/cmd_guard_remove.lua

```diff
@@ -51,12 +58,24 @@ function widget:UnitCommand(unitID, unitDefID, unitTeam, cmdId, cmdParams, cmdOp
 		end
 	end
 
-	doCommandRemove = false
 	return false
 end
 
-function widget:KeyPress(key, modifier, isRepeat)
-	if not isRepeat and (key == KEYSYMS.LSHIFT or key == KEYSYMS.RSHIFT) then
-		doCommandRemove = true
+function widget:Update(dt)
+	updateTime = updateTime + dt
+	if updateTime < 0.25 then
+		return
+	end
+
+	local oldIds = {}
+
+	for i, t in pairs(recentUnits) do
+		if t < spGetGameFrame() - 5 then
+			oldIds[#oldIds + 1] = i
```

Comment:
Why not set `recentUnits[i] = nil` right here? What does the oldIds accomplish?

Reply:
It may be unnecessary, I just have a very strong habit of not removing elements from an array that I'm currently iterating over, it's bitten me too many times in the past. I'll add some comments to that effect

Reply:
Probably a good habit

Reply:
The docs seem to indicate that you can assign nil while iterating: https://www.lua.org/manual/5.1/manual.html#pdf-next